### PR TITLE
(feat) implemented cleanImmediateDus and used it in the immediate call, fixed mocksServer lowercase interaction

### PR DESCRIPTION
**Pull request by Rany615**

### CODE REVIEW

File: mocks/mocksServer.js [hunk 1]
```diff
@@ -63,7 +63,7 @@ function searchInData(data, queries) {
     let foundRecord;
     for (query of queries) {
         foundRecord = data.filter(record => {
-            return JSON.stringify(record).includes(query);
+            return JSON.stringify(record).toLocaleLowerCase().includes(query);
```
Comment (NoamYes) [inline mocks/mocksServer.js:66]: COnsider passing 'en' for locale

File: src/runningMethods/immediate.js [hunk 2]
```diff
@@ -32,20 +32,17 @@ module.exports = async (dataSource, identifiersArray, runUID) => {
...
-        //             dataSource,
-        //             dataObj[dataSource].data,
-        //             domainUser,
-        //             sendLog,
-        //             Auth
-        //         );
-        //     }
-
-        // }
+        if(dataSource != fn.dataSources.aka) {
```
Comment (NoamYes) [inline src/runningMethods/immediate.js:35]: remove comments

File: src/util/cleanDu/cleanImmediateDus.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+const getIrrelevantDus = require('./getIrrelevantDus');
+const deleteDus = require('./deleteDus');
+const { logLevel } = require('../logger');
+const logDetails = require('../logDetails');
+const p =  require('../../config/paths');
+const getRawData = require('../getRawData');
+const moment = require('moment');
+//const Set = require("./collections/set");
+const preRun = require('../preRun');
+//receives dataSource,identifiers an object with 3 possible fields, ...
```
Comment (NoamYes) [inline src/util/cleanDu/cleanImmediateDus.js:11]: standard form of comment 
image

File: src/util/cleanDu/cleanImmediateDus.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+//fetches all possible people with the received identifiers from kartoffel, then for each person
+//iterates over all of their domainUsers and checks if it is consistent with the databases (pncy)
+//if the person is still in the database of that datasource then we continue with no action.
+//otherwise we remove the domainUser from the person in kartoffel
+async function cleanImmediateDus(dataSource, identifiers, sendLog, Auth) {
+
+    //first: get the dataSourcePersons from kartoffel
+    let dataSourcePersons = new Set();
+    if (identifiers.identityCard){
```
Comment (NoamYes) [inline src/util/cleanDu/cleanImmediateDus.js:20]: for in - identifier for both pn and id

File: src/util/cleanDu/cleanImmediateDus.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+const preRun = require('../preRun');
+//receives dataSource,identifiers an object with 3 possible fields, ...
+//fetches all possible people with the received identifiers from kartoffel, then for each person
+//iterates over all of their domainUsers and checks if it is consistent with the databases (pncy)
+//if the person is still in the database of that datasource then we continue with no action.
+//otherwise we remove the domainUser from the person in kartoffel
+async function cleanImmediateDus(dataSource, identifiers, sendLog, Auth) {
+
+    //first: get the dataSourcePersons from kartoffel
+    let dataSourcePersons = new Set();
```
Comment (NoamYes) [inline src/util/cleanDu/cleanImmediateDus.js:19]: no need for two sets

File: src/util/cleanDu/cleanImmediateDus.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+        const resPersonByDU = await Auth.axiosKartoffel.get(`${p().KARTOFFEL_PERSON_API}/domainUser/${identifiers.domainUser}`).catch(err => {
+            sendLog(logLevel.error, logDetails.error.ERR_GET_PERSONS_BY_DU , dataSource, err.message);
+        }); 
+        if(resPersonByDU)
+            dataSourcePersons.add(resPersonByDU.data);
+    }
+    let domainUsers = new Set();
+    for(person of dataSourcePersons){
+        for(domainUser of person.domainUsers){
+            if(domainUsers.has(domainUser.uniqueID)) //already iterated over the domainUser
```
Comment (NoamYes) [inline src/util/cleanDu/cleanImmediateDus.js:44]: consider array with "includes"

File: src/util/cleanDu/cleanImmediateDus.js [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+            else
+                domainUsers.add(domainUser.uniqueID)
+
+            const date = moment(new Date()).format("DD.MM.YYYY__HH.mm");
+            const ds = domainUser.dataSource;
+            const du = domainUser.uniqueID.split("@")[0];
+            const idObj = {domainUser : du}
+            const { data, fileName } = await getRawData(ds, fn.runnigTypes.immediateRun, date, sendLog, idObj);
+            if(!data||data.length == 0){
+                await deleteDus([domainUser.uniqueID], Array.from(dataSourcePersons), ds, sendLog, Auth);
```
Comment (NoamYes) [inline src/util/cleanDu/cleanImmediateDus.js:55]: get irrelevantDus by upn

File: src/util/logDetails.js [hunk 8]
```diff
@@ -26,6 +26,7 @@ messageLog = {
         ERR_NOT_FOUND_IN_RAW_DATA: 'The person with identifier %s from %s does not exists in raw data',
         ERR_UNRECOGNIZED_DATA_SOURCE: 'Unrecognized dataSource at %s, the dataSource is: %s, the record is: %s',
         ERR_GET_PERSONS_BY_DU: 'Failed to get persons by domain user of %s from kartoffel. The error is: %s',
```
Comment (NoamYes) [inline src/util/logDetails.js:28]: ADD DATASOURCE

**Review by NoamYes — COMMENTED** [review]
Good luck

File: src/util/cleanDu/cleanImmediateDus.js (hunk 4 above)
Comment (Rany615) [reply src/util/cleanDu/cleanImmediateDus.js:20]: cant do "for await in" like the "for await of"

File: src/util/cleanDu/cleanImmediateDus.js (hunk 6 above)
Comment (Rany615) [reply src/util/cleanDu/cleanImmediateDus.js:44]: changed the people set to array, but when checking for uniqueness the set of domain users seems to be more fitting than it to be an array, should i change it regardless?

File: mocks/mocksServer.js (hunk 1 above)
Comment (Rany615) [reply mocks/mocksServer.js:66]: changed to "toLowerCase()" instead

File: src/util/cleanDu/cleanImmediateDus.js [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+ * @param {string} dataSource 
+ * @param {struct} identifiers an object with 3 possible fields
+ * @param {*} sendLog 
+ * @param {*} Auth 
+ */
+async function cleanImmediateDus(dataSource, identifiers, sendLog, Auth) {
+
+    //first: get the dataSourcePersons from kartoffel
+    let dataSourcePersons = [];
+    for (identifier in identifiers){
```
Comment (NoamYes) [inline src/util/cleanDu/cleanImmediateDus.js:25]: check that out

File: src/util/cleanDu/cleanImmediateDus.js [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+            const ds = domainUser.dataSource;
+            const du = domainUser.uniqueID.split("@")[0];
+            const idObj = {domainUser : du}
+            const { data, fileName } = await getRawData(ds, fn.runnigTypes.immediateRun, date, sendLog, idObj);
+            //ads datasource exception
+            let irrelevant = []
+            if (ds == fn.dataSources.ads) {
+                irrelevant = getIrrelevantDus(data,dataSourcePersons,ds,sendLog,Auth);
+            }
+            if(!data||data.length == 0||irrelevant.length!=0){
```
Comment (NoamYes) [inline src/util/cleanDu/cleanImmediateDus.js:66]: !data || data.length == 0 || irrelevant.length != 0

**Review by NoamYes — COMMENTED** [review]
Minor comments

**Pull request merged**